extensions/blocks/simple-payments/utils.js

```diff
@@ -21,9 +22,50 @@ export const decimalPlaces = number => {
 	return Math.max( 0, ( match[ 1 ] ? match[ 1 ].length : 0 ) - ( match[ 2 ] ? +match[ 2 ] : 0 ) );
 };
 
-export const formatPrice = ( price, currency, withSymbol = true ) => {
+const getNavigatorLanguage = () => {
+	if ( navigator.languages && navigator.languages.length ) {
+		return navigator.languages[ 0 ];
+	}
+	return navigator.userLanguage || navigator.language || navigator.browserLanguage || 'en';
+};
+
+// Legacy method of displaying prices.
+export const formatPriceFallback = ( price, currency, withSymbol = true ) => {
 	const { precision, symbol } = getCurrencyDefaults( currency );
 	const value = price.toFixed( precision );
 	// Trim the dot at the end of symbol, e.g., 'kr.' becomes 'kr'
 	return withSymbol ? `${ value } ${ trimEnd( symbol, '.' ) }` : value;
 };
+
+// Display prices using Intl.NumberFormat - supported in 95.75% of browsers as of Oct 2020.
+const formatPriceIntlNumberFormat = ( price, currency, withSymbol = true ) => {
+	const { siteLocale } = select( 'core/block-editor' ).getSettings();
+	const tryLocales = [ siteLocale, getNavigatorLanguage(), 'en-US' ];
+
+	let formatOptions = {};
+	if ( withSymbol ) {
+		formatOptions = { style: 'currency', currency };
+	}
+
+	// We're not 100% certain that the siteLocale or the navigatorLanguage line up with Intl.NumberFormat ("best effort" API).
+	// Try them in order, fallback to "en-US".
+	let locale;
+	for ( locale of tryLocales ) {
+		try {
+			return Intl.NumberFormat( locale, formatOptions ).format( price );
+		} catch {
+			continue;
+		}
+	}
+
+	// "Shouldn't" reach here - maybe Intl.Numberformat rejected the currency. Fallback.
+	return formatPriceFallback( price, currency, withSymbol );
+};
+
+// Format a price with Intl.NumberFormat if available, but fall back to old method if not.
+export const formatPrice = ( price, currency, withSymbol = true ) => {
+	if ( window.Intl && 'function' === typeof Intl.NumberFormat ) {
+		return formatPriceIntlNumberFormat( price, currency, withSymbol );
+	}
+	return formatPriceFallback( price, currency, withSymbol );
```

Comment:
Can't this be part of a generic `formatPriceIntlNumberFormat` which will return `formatPriceFallback( price, currency, withSymbol );` if not `window.Intl && 'function' === typeof Intl.NumberFormat`?

Reply:
I'm not sure what you're referring to - is it this? 
* [fix/currency-display c9aa474b6] Inline formatPrice selector function
Personally, I prefer the previous version that has a slightly simpler function at the trade-off of adding one more function.

Reply:
Yes, that is exactly what I meant. 

> Personally, I prefer the previous version that has a slightly simpler function at the trade-off of adding one more function.

From a reading point (while reviewing), I felt it weird but feel free to revert using your version in that case!